### PR TITLE
Implement POST /events Fixes #21

### LINKED ISSUE #21: Implement `POST /events`
## Tasks:

- Implement `POST /events`
- Add documentation and unit testing.

### PR DESCRIPTION
**Pull request by dilarafirtina**
Fixes #21

### CODE REVIEW

File: src/controllers/__tests__/events.test.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ const ngos = [
 
 const volunteers = [
   {
-    _id: '1',
+    id: '1',
     firstName: 'Dilara',
     lastName: 'Fırtına',
```
Comment (yamanrajab90) [inline src/controllers/__tests__/events.test.js:70]: test passed successfully, the Last name includes  The character U+0131 "ı" could be confused with the character U+0069 "i", which is more common in source code

File: src/controllers/__tests__/events.test.js [hunk 2]
```diff
@@ -96,3 +96,69 @@ describe("Testing events for routes doesn't require auth controls", () => {
...
+          return err;
+        }
+        expect(req.user.id).toBe(ngos.id);
+        expect(res.body.data.avatar).toEqual(events[0].avatar);
+        expect(res.body.data.description).toEqual(events[0].description);
+        expect(res.body.data.location).toEqual(events[0].location);
+        expect(res.body.data.launchDate).toEqual(events[0].launchDate);
+        expect(res.body.data.ngoId).toEqual(events[0].ngo);
+        expect(res.body.data.topic).toEqual(events[0].topic);
+        done();
```
Comment (yamanrajab90) [inline src/controllers/__tests__/events.test.js:124]: Awesome work! 
excellent testing strategy.

File: src/routes/events.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
 const router = require('express').Router();
-
+const { validate } = require('../middlewares/bodyValidator');
 const eventsControllers = require('../controllers/events');
-//  TODO: AUTH middleware will be added .
+const { validateAddEvent } = require('../middlewares/validatorSchemas');
+const { isAuth } = require('../middlewares/auth');
```
Comment (yamanrajab90) [inline src/routes/events.js:5]: good declaration in accordance with eslint.
Great job/

**Review by yamanrajab90 — APPROVED** [review]

File: src/controllers/events.js [hunk 4]
```diff
@@ -1,9 +1,37 @@
...
+  newEvent.location = req.body.location;
+  newEvent.launchDate = req.body.launchDate;
+  newEvent.ngo = req.user._id;
+  newEvent.topic = req.body.topic;
+  if (req.body.pendingApplicantsId)
+    newEvent.pendingApplicants.push(req.body.pendingApplicantsId);
+  if (req.body.approvedApplicantsId)
+    newEvent.approvedApplicants.push(req.body.approvedApplicantsId);
+  if (req.body.declinedApplicantsId)
+    newEvent.declinedApplicants.push(req.body.declinedApplicantsId);
```
Comment (muslimomar) [inline src/controllers/events.js:31]: The event should have empty `pendingApplicants, approved, declined`, so just leave them as their default values.

File: src/controllers/events.js [hunk 5]
```diff
@@ -1,9 +1,37 @@
...
+    return next(new ErrorResponse('No NGO found', 404));
+  }
+  // Create new event
+  const newEvent = new Event();
+  newEvent.avatar = req.body.avatar;
+  newEvent.description = req.body.description;
+  newEvent.location = req.body.location;
+  newEvent.launchDate = req.body.launchDate;
+  newEvent.ngo = req.user._id;
+  newEvent.topic = req.body.topic;
```
Comment (muslimomar) [inline src/controllers/events.js:25]: You could use a shorthand for these operations: 

`const newEvent = new Event(req.body);` or `const newEvent = await Event.create(req.body)`

File: src/controllers/__tests__/events.test.js [hunk 6]
```diff
@@ -4,22 +4,22 @@ const app = require('../../app');
 jest.setTimeout(10000);
 
 const user1 = {
-  _id: '1',
+  id: '1',
```
Comment (muslimomar) [inline src/controllers/__tests__/events.test.js:7]: Why are you removing the underscore? 🤔

File: src/controllers/__tests__/events.test.js [hunk 7]
```diff
@@ -96,3 +96,69 @@ describe("Testing events for routes doesn't require auth controls", () => {
...
+      user: {
+        id: user2.id,
+      },
+    };
+    request(app)
+      .post('/api/events/')
+      .expect(403)
+      .expect('Content-Type', 'application/json; charset=utf-8')
+      .expect({
+        message: 'You are not authorized to delete this event',
```
Comment (muslimomar) [inline src/controllers/__tests__/events.test.js:160]: The message says: 'You are not authorized to delete this event'. delete or add an event? 🤣

**Review by muslimomar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**